### ISSUE TITLE
Regression in 0.83.0: ParenthesesAsGroupedExpression for heredoc

### ISSUE THREAD

**Issue by jaredbeck** [rubocop-hq/rubocop#7962]
## Expected behavior

The following input should not produce an offense.

```
def users
  ::User.redacted(@redacted.redacted).where(
    <<-SQL
      exists (
        select 1
        from redacted
        where (
          users.id = redacted.redacted_id
          or users.id = redacted.redacted2_id
        )
      )
    SQL
  )
end
```

## Actual behavior

The above input produced the following offense.

```
app/redacted/redacted/redacted.rb:34:12: W: Lint/ParenthesesAsGroupedExpression: (...) interpreted as grouped expression.
            <<-SQL
           ^
```

## Steps to reproduce the problem

Place the above input in a file and run rubocop 0.83.0 on it.

Confirm regression by running the same experiment in 0.82 and *not* producing an offense.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
0.83.0 (using Parser 2.7.1.2, running on ruby 2.6.6 x86_64-darwin19)
```

**Comment by jaredbeck** [comment]
Thanks Koichi and Bozhidar! That was fast.

**Comment by bbatsov** [comment]
You're welcome!
